app/src/main/java/com/example/habitsmasher/ui/dashboard/HabitListFragment.java

```diff
@@ -256,8 +262,20 @@ public void addHabitToDatabase(String title, String reason, Date date, DaysTrack
      */
     public void updateAfterEdit(String newTitle, String newReason, Date newDate, int pos,
                                 DaysTracker tracker) {
-        _habitList.editHabitInDatabase(newTitle, newReason, newDate, tracker, pos, _user.getUsername());
+        _habitList.editHabitInDatabase(newTitle, newReason, newDate, tracker, pos, _user.getId());
         _habitList.editHabitLocal(newTitle, newReason, newDate, tracker, pos);
         _habitItemAdapter.notifyItemChanged(pos);
     }
+
+    @NonNull
+    private User getCurrentUser() {
+        SharedPreferences sharedPref = _context.getSharedPreferences(USER_DATA_PREFERENCES_TAG, Context.MODE_PRIVATE);
+
+        String username = sharedPref.getString("username", "user");
+        String userId = sharedPref.getString("userId", "id");
+        String email = sharedPref.getString("email", "email");
+        String password = sharedPref.getString("password", "password");
+
+        return new User(userId, username, email, password);
```

Comment:
Could this be taken out and used as a general function? I feel this can be extracted.